cmds/rush/readline.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/chzyer/readline"
+)
+
+func usage(w io.Writer) {
+	io.WriteString(w, "commands:\n")
+	io.WriteString(w, completer.Tree("    "))
+}
+
+// Function constructor - constructs new function for listing given directory
+func listFiles(path string) func(string) []string {
+	return func(line string) []string {
```

Comment:
Sorry @rminnich but I don't understand the use of this anonymous function inside the body of `listFiles`. Wouldn't be easier to just make `listFiles` return `[]string` and remove the anonymous function inside `listFiles`? I don't see any mention to the parameter passed to it either so I must suppose it has no use yet.